toothpick-runtime/src/main/java/toothpick/Toothpick.java

```diff
@@ -145,6 +164,7 @@ private static Scope openScope(Object name, boolean shouldCheckMultipleRootScope
       // if there was already a scope by this name, we return it
       scope = previous;
     } else if (shouldCheckMultipleRootScopes) {
```

Comment:
can we rename `shouldCheckMultipleRootScopes` to `isRootScope`?

`shouldCheckMultipleRootScopes` does not mean that the provided scope is a root scope. It is true that the value will be true if the scope is a root scope, so i think it is better to change the name.

Reply:
Yes and ifnI remember correctly the check itself has some branches of its if logic that don't make any sense. This is highlighted in the code coverage of the class as well.